### PR TITLE
Feature/create database

### PR DESCRIPTION
**Pull request by andela-fabolaji**
### What does this PR do?
Build api backend

### Description of Task to be completed?
- Bootstrap project - Basic environment setup
- Setup user authentication
- Setup the following endpoints: `role`, `user`, `type`, and `document`
- Create the following access levels: `public`, `private`, `role`
- Write unit tests for all endpoints

### CODE REVIEW

**Comment by andela-aolaniran** [comment]
Nice job. I like the creativity and abstraction employed. I feel it should be just fine after you address all issues we spotted during the review. Also, try to return appropriate status codes. Once again, nice one.

File: server/db/seeds/Users.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+  {
+    firstName: 'Shalom',
+    lastName: 'Ayidu',
+    email: 'shalom.ayidu@youdoc.com',
+    username: 'shalom',
+    password: 'password',
+    role: 'fellow',
+    createdAt: new Date(),
+    updatedAt: new Date()
+  }, {
```
Comment (andela-kakpobome) [inline server/db/seeds/Users.js]: ### Nitpick

The opening brace should be on the next line, we already discused this.

File: server/tests/api/a.role.spec.js [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+          done();
+        });
+    });
+    it('PUT - Should be able to update role with valid title', (done) => {
+      request
+        .put('/roles/2')
+        .set('authorization', admin.token)
+        .send({ title: 'super admin' })
+        .end((err, res) => {
+          // console.log(res.body);
```
Comment (andela-kakpobome) [inline server/tests/api/a.role.spec.js]: ### Critical

There should be no `console.logs` in production.

File: server/tests/api/a.role.spec.js [hunk 3]
```diff
@@ -0,0 +1,329 @@
...
+          done();
+        });
+    });
+    it('POST - Should not be able to create duplicate role', (done) => {
+      request
+        .post('/roles')
+        .set('authorization', admin.token)
+        .send(invalidRoles[0])
+        .end((err, res) => {
+          expect(res.status).to.equal(400);
```
Comment (andela-kakpobome) [inline server/tests/api/a.role.spec.js]: ### Critical

This is an invalid response for duplicate resources.

**Pull request merged**